model.go

```diff
@@ -46,7 +47,18 @@ func (m *Model) ID() interface{} {
 // IDField returns the name of the DB field used for the ID.
 // By default, it will return "id".
 func (m *Model) IDField() string {
-	field, ok := reflect.TypeOf(m.Value).Elem().FieldByName("ID")
+	modelType := reflect.TypeOf(m.Value)
+
+	// remove all indirections
+	for modelType.Kind() == reflect.Slice || modelType.Kind() == reflect.Ptr || modelType.Kind() == reflect.Array {
+		modelType = modelType.Elem()
+	}
+
+	if modelType.Kind() == reflect.String {
```

Comment:
Can model be a string?

Reply:
Yes, there are several test cases for that, e.g. https://github.com/gobuffalo/pop/blob/master/executors_test.go#L1506